### PR TITLE
Upgrade dependencies and migrate to zimscraperlib 3.3.2

### LINKED ISSUE #175: Upgrade python-scraperlib to 3.x, including CLI support for description / long_description flags
See https://github.com/openzim/overview/issues/21

### LINKED ISSUE #188: Rework or remove the `playlist_mode`
In this scraper, we have a `playlist_mode` which allows to create a ZIM per playlist found in a given Youtube user / channel.

This mode is convenient to create many ZIMs at once, but it poses an issue in terms of metadata quality since titles, descriptions, ... are automatically sourced from Youtube.

With the move to scraperlib 3.x, the creation of ZIMs with invalid title, description, ... will fail. Unfortunately, this check is done only at the end of the scraping since we still use the "zimwriterfs" mode with `make_zim_file` at the end of the scraper, after all videos have been downloaded and reencoded.

We should either:
- invest time to rework the functionality, compute only valid metadata (even if this is deemed to be limited) and check this validity before starting any content processing
- remove the functionality since we won't use it on the Zimfarm, do not have sufficient resources to do the rework and the rework quality will anyway provide results of limited quality or require vast efforts (e.g. computing a 30 chars Title automatically is not an easy feat)

This is a blocker for #175 in fact (or we accept to have a functionality which will not work in 90% of the cases)

### PR DESCRIPTION
**Pull request by dan-niles**
### Changes
- Migrate to zimscraperlib 3.3.2
- Upgrade Python dependencies, including migration to Python 3.12
- Validate ZIM metadata (tags, title, description, long_description) as early as possible
- Added new CLI arguments: `long_description` and `disable_metadata_check`
- Added `disable_metadata_check` to `playlist mode`

Fix #175
Fix #188

### CODE REVIEW

**Review by benoit74 — CHANGES_REQUESTED** [review]
This is very good work, thank you!

Please add support for `long_description` in the `playlists` mode as well, not sure it is very valuable but at least we have it available, and it is mostly straightforward AFAIK:
- add arg to https://github.com/openzim/youtube/blob/main/src/youtube2zim/playlists/entrypoint.py
- update list at https://github.com/openzim/youtube/blob/d930c98a7a5247dba5a04ea2054888e44de9bcf2/src/youtube2zim/playlists/scraper.py#L142
- do not forgot update list at https://github.com/openzim/youtube/blob/d930c98a7a5247dba5a04ea2054888e44de9bcf2/src/youtube2zim/playlists/entrypoint.py#L78 to raise exception if `--long-description` is passed instead of `--playlists-long-description`

Few inline comments also indicate more minor changes needed.

Glad you achieved to make it worked properly without any help, it wasn't an easy thing ^^

File: openzim.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+[files.assets.config]
+target_dir="src/youtube2zim/templates/assets"
+execute_after=[
+    "fix_ogvjs_dist .",
+]
+
+[files.assets.actions."video.js"]
+action="extract_all"
+source="https://github.com/videojs/video.js/releases/download/v7.8.1/video-js-7.8.1.zip"
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply openzim.toml]: Changed in [76da367](https://github.com/openzim/youtube/pull/202/commits/76da3679a7b33284f940e644f255212595400042)

File: openzim.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[files.assets.actions."chosen.jquery.js"]
+action="extract_all"
+source="https://github.com/harvesthq/chosen/releases/download/v1.8.7/chosen_v1.8.7.zip"
+target_dir="chosen"
+remove = ["docsupport","chosen.proto.*","*.html","*.md"]
+
+[files.assets.actions."jquery.min.js"]
+action="get_file"
+source="https://code.jquery.com/jquery-3.5.1.min.js"
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply openzim.toml]: Changed in [76da367](https://github.com/openzim/youtube/pull/202/commits/76da3679a7b33284f940e644f255212595400042)

File: openzim.toml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+[files.assets.actions."videojs-ogvjs.js"]
+action="extract_items"
+source="https://github.com/hartman/videojs-ogvjs/archive/v1.3.1.zip"
+zip_paths=["videojs-ogvjs-1.3.1/dist/videojs-ogvjs.js"]
+target_paths=["videojs-ogvjs.js"]
+
+[files.assets.actions."webp-hero.polyfills.js"]
+action="get_file"
+source="https://unpkg.com/webp-hero@0.0.0-dev.26/dist-cjs/polyfills.js"
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply openzim.toml]: Changed in [76da367](https://github.com/openzim/youtube/pull/202/commits/76da3679a7b33284f940e644f255212595400042)

File: openzim.toml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+target_file="polyfills.js"
+
+[files.assets.actions."webp-hero.bundle.js"]
+action="get_file"
+source="https://unpkg.com/webp-hero@0.0.0-dev.26/dist-cjs/webp-hero.bundle.js"
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply openzim.toml]: Changed in [76da367](https://github.com/openzim/youtube/pull/202/commits/76da3679a7b33284f940e644f255212595400042)

File: pyproject.toml [hunk 5]
```diff
@@ -1,54 +1,50 @@
...
-  "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
-]
 dependencies = [
   "python-slugify==3.0.3",
   "yt-dlp",                      # youtube-dl should be updated as frequently as possible
-  "python-dateutil==2.8.0",
-  "jinja2>=2.11,<3.0",
+  "python-dateutil==2.9.0.post0",
+  "jinja2==3.1.4",
   "MarkupSafe==2.0.1",           # jinja2 dependency (https://github.com/pallets/markupsafe/issues/284)
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply pyproject.toml]: Removed in [7d66802](https://github.com/openzim/youtube/pull/202/commits/7d668023b6f45eab6fa0142dd419791ee4dcfd65)

**Review by benoit74 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**